src/services/services.ts

```diff
@@ -1635,6 +1635,77 @@ module ts {
         sourceFile.scriptSnapshot = scriptSnapshot;
     }
 
+    export function transpile(input: string, compilerOptions?: CompilerOptions, fileName?: string, syntaxErrors?: Diagnostic[]): string {

```

Comment:
We need some comments on what this function is about and the constraints of using it.